### PR TITLE
LC-2307 Post registration user profile setup-3

### PR DESCRIPTION
**Pull request by neeraj-jain-cbo**

### CODE REVIEW

File: src/main/java/uk/gov/cabinetoffice/csl/service/IdentityService.java [hunk 1]
```diff
@@ -88,12 +111,22 @@ public Identity getIdentityForEmail(String email) {
         return identityRepository.findFirstByEmailEqualsIgnoreCase(email);
     }
 
-    public Identity getIdentityForEmailAndActiveFalse(String email) {
+    public Identity getActiveIdentityForEmail(String email) {
+        return identityRepository.findFirstByActiveTrueAndEmailEqualsIgnoreCase(email);
+    }
+
+    public Identity getInActiveIdentityForEmail(String email) {
```
Comment (SavvasCO) [inline src/main/java/uk/gov/cabinetoffice/csl/service/IdentityService.java:118]: I think `InActive` should be `Inactive`.

**Review by SavvasCO — COMMENTED** [review]
A couple of comments

File: src/main/java/uk/gov/cabinetoffice/csl/service/IdentityService.java (hunk 1 above)
Comment (neeraj-jain-cbo) [reply src/main/java/uk/gov/cabinetoffice/csl/service/IdentityService.java]: Done

File: src/test/java/uk/gov/cabinetoffice/csl/handler/CustomAuthenticationFailureHandlerTest.java [hunk 2]
```diff
@@ -70,7 +70,7 @@ public void shouldSetErrorToDeactivatedExpiredOnDeactivationExpired() throws IOE
     public void shouldSetErrorToDeactivatedOnAccountDeactivatedAndPendingReactivationExists()
             throws Exception {
         HttpServletResponse response = executeHandler("Pending reactivation exists for user");
-        verify(response).sendRedirect("/login?error=pending-reactivation&requestedAt=01 Feb 2024 15:30:20");
+        verify(response).sendRedirect("/login?error=pending-reactivation&requestedAt=01 Feb 2024 15:30:20&reactivationLinkExpiry=02 Feb 2024 15:30:20");
```
[the remark this answers is not in the document]
Comment (neeraj-jain-cbo) [reply src/test/java/uk/gov/cabinetoffice/csl/handler/CustomAuthenticationFailureHandlerTest.java]: Done

**Review by jaylco — APPROVED** [review]

**Pull request merged**